app/models/manageiq/providers/redhat/infra_manager/admin_ui.rb

```diff
@@ -0,0 +1,96 @@
+module ManageIQ::Providers::Redhat::InfraManager::AdminUI
+  extend ActiveSupport::Concern
+
+  require 'openssl'
+  require 'ovirtsdk4'
+  require 'uri'
+  require 'json'
+
+  def queue_generate_admin_ui_url
+    task_opts = {
+      :action => "Generate oVirt Admin UI URL for EMS"
+    }
+
+    queue_opts = {
+      :class_name  => self.class.name,
+      :instance_id => id,
+      :method_name => "generate_admin_ui_url",
+      :priority    => MiqQueue::HIGH_PRIORITY,
+      :role        => "ems_operations",
+      :zone        => my_zone,
+      :args        => []
+    }
+
+    MiqTask.generic_action_with_callback(task_opts, queue_opts)
+  end
+
+  def generate_admin_ui_url
+    $rhevm_log.info("Generating oVirt Admin UI URL for EMS with identifier '#{id}'.")
+
+    # Get the credentials of the provider:
+    user = authentication_userid(:default)
+    password = authentication_password(:default)
+
+    # Create the HTTP client:
+    insecure = default_endpoint.verify_ssl == OpenSSL::SSL::VERIFY_NONE,
+    ca_certs = default_endpoint.certificate_authority
+    ca_certs = nil if ca_certs.blank?
+    ca_certs = [ca_certs] if ca_certs
+    client = OvirtSDK4::HttpClient.new(
+      :insecure => insecure,
+      :ca_certs => ca_certs,
+      :log      => $rhevm_log
+    )
+
+    # Request new SSO token using the provider credentials:
+    request = OvirtSDK4::HttpRequest.new
+    request.method = :GET
+    request.url = URI::HTTPS.build(
+      :host  => default_endpoint.hostname,
+      :port  => default_endpoint.port,
+      :path  => "/ovirt-engine/sso/oauth/token",
+      :query => URI.encode_www_form(
+        :grant_type => "password",
+        :username   => user,
+        :password   => password,
+        :scope      => "ovirt-app-admin ovirt-app-portal"
+      )
+    ).to_s
+    request.headers = {
+      "Accept"     => "application/json",
+      "User-Agent" => "manageiq-providers-ovirt / API version #{api_version}"
```

Comment:
The typical format of the `User-Agent` header is `name/version`, without spaces or without additional text before the version number. I'd suggest `"manageiq-providers-ovirt/#{ManageIQ::Providers::Ovirt::VERSION}"`, for example.

Note that I am not sure if we ever update that `VERSION`, we should start doing it, probably.

Reply:
OK, I'll change the header value to `manageiq-providers-ovirt/<version>`.

Reply:
> Note that I am not sure if we ever update that `VERSION`, we should start doing it, probably.

This version is also used in `manageiq-providers-ovirt.gemspec` so it represents the oVirt Provider Ruby gem version, we can use that.